frontend/src/lib/components/CommandPalette/commandPaletteLogic.ts

```diff
@@ -0,0 +1,681 @@
+        togglePalette: true,
+        setInput: (input: string) => ({ input }),
+        onArrowUp: true,
+        onArrowDown: (maxIndex: number) => ({ maxIndex }),
+        onMouseEnterResult: (index: number) => ({ index }),
+        onMouseLeaveResult: true,
+        executeResult: (result: CommandResult) => ({ result }),
+        activateFlow: (flow: CommandFlow | null) => ({ flow }),
+        registerCommand: (command: Command) => ({ command }),
+        deregisterCommand: (commandKey: string) => ({ commandKey }),
+        setCustomCommand: (commandKey: string) => ({ commandKey }),
+        deregisterAllWithMatch: (keyPrefix: string) => ({ keyPrefix }),
+    },
+    reducers: {
+        isPaletteShown: [
+            false,
+            {
+                hidePalette: () => false,
+                showPalette: () => true,
+                togglePalette: (previousState) => !previousState,
+            },
+        ],
+        keyboardResultIndex: [
+            0,
+            {
+                setInput: () => 0,
+                executeResult: () => 0,
+                activateFlow: () => 0,
+                onArrowUp: (previousIndex) => (previousIndex > 0 ? previousIndex - 1 : 0),
+                onArrowDown: (previousIndex, { maxIndex }) => (previousIndex < maxIndex ? previousIndex + 1 : maxIndex),
+            },
+        ],
+        hoverResultIndex: [
+            null as number | null,
+            {
+                onMouseEnterResult: (_, { index }) => index,
+                onMouseLeaveResult: () => null,
+                onArrowUp: () => null,
+                onArrowDown: () => null,
+                activateFlow: () => null,
+            },
+        ],
+        input: [
+            '',
+            {
+                setInput: (_, { input }) => input,
+                activateFlow: () => '',
+                executeResult: () => '',
+            },
+        ],
+        activeFlow: [
+            null as CommandFlow | null,
+            {
+                activateFlow: (_, { flow }) => flow,
+            },
+        ],
+        rawCommandRegistrations: [
+            {} as CommandRegistrations,
+            {
+                registerCommand: (commands, { command }) => {
+                    return { ...commands, [command.key]: command }
+                },
+                deregisterCommand: (commands, { commandKey }) => {
+                    const cleanedCommands = { ...commands }
+                    delete cleanedCommands[commandKey]
+                    return cleanedCommands
+                },
+            },
+        ],
+    },
+
+    listeners: ({ actions, values }) => ({
+        showPalette: () => {
+            window.posthog?.capture('palette shown')
+        },
+        togglePalette: () => {
+            if (values.isPaletteShown) window.posthog?.capture('palette shown')
+        },
+        executeResult: ({ result }: { result: CommandResult }) => {
+            if (result.executor === true) {
```

Comment:
nit: `=== true` seems like a code smell, though I guess I understand the intent here. Probably nothing really to improve.